app/component/favourites/FavouriteRouteListContainer.js

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import Relay from 'react-relay';
+import NextDeparturesList, {
+   relayFragment as NextDeparturesListRelayFragment,
+} from '../departure/NextDeparturesList';
+import NoPositionPanel from '../front-page/no-position-panel';
+
+import { getDistanceToNearestStop } from '../../util/geo-utils';
+
+import connectToStores from 'fluxible-addons-react/connectToStores';
+
+const getNextDepartures = (routes, lat, lon) => {
+  const nextDepartures = [];
+  const seenDepartures = {};
+
+  for (const route of routes) {
+    for (const pattern of route.patterns) {
+      const closest = getDistanceToNearestStop(lat, lon, pattern.stops);
+      const keepStoptimes = [];
+
+      for (const stoptime of closest.stop.stoptimes) {
+        const seenKey = `${stoptime.pattern.route.gtfsId}:${stoptime.pattern.headsign}`;
+        const isSeen = seenDepartures[seenKey];
+        const isFavourite =
+          stoptime.pattern.route.gtfsId === route.gtfsId &&
+          stoptime.pattern.headsign === pattern.headsign;
+
+        if (!isSeen && isFavourite) {
+          keepStoptimes.push(stoptime);
+          seenDepartures[seenKey] = true;
+        }
+      }
+
+      for (const stoptime of keepStoptimes) {
+        nextDepartures.push({
+          distance: closest.distance,
+          stoptime,
+        });
+      }
+    }
+  }
+
+  return nextDepartures;
+};
+
+function FavouriteRouteListContainer({ location, currentTime, searching, routes }) {
+  if (location) {
+    return (
+      <NextDeparturesList
+        departures={getNextDepartures(routes, location.lat, location.lon)}
+        currentTime={currentTime.unix()}
+      />
+    );
+  } else if (searching) {
+    return <div className="spinner-loader" />;
+  }
+  return <NoPositionPanel />;
+}
+
+FavouriteRouteListContainer.propTypes = {
+  routes: React.PropTypes.array,
+  currentTime: React.PropTypes.object,
+  searching: React.PropTypes.bool,
+  location: React.PropTypes.oneOfType([React.PropTypes.object, React.PropTypes.bool]).isRequired,
+};
+
+// TODO: This should be moved above in the component hierarchy
+const FavouriteRouteListContainerWithTime = connectToStores(
```

Comment:
Maybe this could be split into multiple wrappers with separate time and position?